packages/db/src/core/cli/commands/link/index.ts

```diff
@@ -1,36 +1,76 @@
 import type { AstroConfig } from 'astro';
+import { homedir } from 'node:os';
 import { mkdir, writeFile } from 'node:fs/promises';
 import { bgRed, cyan } from 'kleur/colors';
+import { slug } from 'github-slugger';
 import prompts from 'prompts';
 import type { Arguments } from 'yargs-parser';
 import { PROJECT_ID_FILE, getSessionIdFromFile } from '../../../tokens.js';
 import { getAstroStudioUrl } from '../../../utils.js';
 import { MISSING_SESSION_ID_ERROR } from '../../../errors.js';
+import { basename } from 'node:path';
+import ora from 'ora';
 
 export async function cmd({ flags }: { config: AstroConfig; flags: Arguments }) {
-	const linkUrl = new URL(getAstroStudioUrl() + '/auth/cli/link');
 	const sessionToken = await getSessionIdFromFile();
 	if (!sessionToken) {
 		console.error(MISSING_SESSION_ID_ERROR);
 		process.exit(1);
 	}
-	let body = { id: flags._[4] } as {
-		id?: string;
-		projectIdName?: string;
-		workspaceIdName?: string;
-	};
-	if (!body.id) {
-		const workspaceIdName = await promptWorkspaceName();
-		const projectIdName = await promptProjectName();
-		body = { projectIdName, workspaceIdName };
+	await promptBegin();
+	const isLinkExisting = await promptLinkExisting();
+	if (isLinkExisting) {
+		const projectId = await promptExistingProjectName();
+		return await linkProject(projectId);
+	}
+
+	const isLinkNew = await promptLinkNew();
+	if (isLinkNew) {
+		const newProjectName = await promptNewProjectName();
+		const newProjectRegion = await promptNewProjectRegion();
+		const newProjectId = await createNewProject({name: newProjectName, region: newProjectRegion});
+		// TODO(fks): Actually listen for project creation before continuing
+		// This is just a dumb spinner that roughly matches database creation time.
+		const spinner = ora('Creating new project...');
+		await new Promise((r) => setTimeout(r, 4000));
+		spinner.succeed('Project created!');
+		return await linkProject(newProjectId);
+	}
+}
+
+async function linkProject(id: string) {
+	await mkdir(new URL('.', PROJECT_ID_FILE), { recursive: true });
+	await writeFile(PROJECT_ID_FILE, `${id}`);
+	console.info('Project linked.');
+}
+
+export async function createNewProject({name, region}: {name: string, region: string}): Promise<string> {
+	const linkUrl = new URL(getAstroStudioUrl() + '/api/cli/projects.create');
+	const response = await fetch(linkUrl, {
+		method: 'POST',
+		headers: {
+			Authorization: `Bearer ${await getSessionIdFromFile()}`,
+			'Content-Type': 'application/json',
+		},
+		body: JSON.stringify({ name, region }),
+	});
+	if (!response.ok) {
+		console.error(`Failed to create project: ${response.status} ${response.statusText}`);
```

Comment:
Can we standardize to the `{ success: boolean, value: JSON }` format for reading errors? I worry `statusText` could log errors that aren't human-readable

Reply:
ah good catch, I think that already is the response format and this is just checking for the case where our entire API fails and you just get back an unhandled 500. But, we should also have proper handling for the handled error case.